README.md
# g3grey
G3Grey Theme for Gallery 3

The theme is based on the original Grey Dragon theme.

http://codex.galleryproject.org/Gallery3:Themes:greydragon.html

https://blog.dragonsoft.us/gallery-3/

G3Grey is developed for Gallery the Revival, v3.1.5.

https://galleryrevival.com/


The latest release could be found at:
https://forum.g2soft.net/viewforum.php?f=36


changelog.txt
=== G3 Grey Theme ===
G3 Grey Theme - a custom theme for Gallery 3

This theme is designed and built by David Yin, https://www.yinfor.com
Copyright (C) 2023 David Yin

Based on the Grey Dragon Theme, which was designed and built by Serguei Dosyukov,
whose blog you will find at http://blog.dragonsoft.us/
Copyright (C) 2009-2012 Serguei Dosyukov

Tested up to: Gallery the Revival 3 3.1.5


This program is free software; you can redistribute it and/or modify it under
the terms of the GNU General Public License as published by the Free Software
Foundation; either version 2 of the License, or (at your option) any later
version. This program is distributed in the hope that it will be useful, but
WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
details. You should have received a copy of the GNU General Public License
along with this program; if not, write to the Free Software Foundation, Inc.,
51 Franklin Street Fifth Floor, Boston, MA  02110-1301, USA.

=== Open issues ===
- RTL layout support
                            
=== Changelog ===
Version 5.0.0
- Remove the short open tags
- Fix the related modules

=== Before David Yin ===
version 4.0.2
- Incompatability fix for PHP5.4 and below causing crash and blank page. http://galleryproject.org/forum_theme_greydragon?page=48#comment-408923
- Fix to allow thumbnail rotation update when operation is performed. http://galleryproject.org/forum_theme_greydragon?page=48#comment-408993
- Cycle JS: Added fadeout support. Fixed issue in root page when "Click to Enter" link would disappear with some effects

version 4.0.1
- Code converted to use latest jQuery library
- Added support for RWInfo meta data injection
- CSS: Fixes a phenomenon where dialog content not setting proper width in FF/Ch
- CSS: Fixes a phenomenon where context menu shows empty space above first item
- Added support for g-movie class in thumbs
- Admin: Better help translation
- Logic to load frame template optimized to prevent exception
- thumb_top()/thumb_botom() call moved inside div.g-thumbslide per Dave's request
- Markup cleanup for W3C validation
- CSS: small layout size adjustments in Simple Frame Pack
- Photo Page: fixed issue with image preload logic when one of the items is an album
- Added code protection to paginator when item is not well defined - missing ->url() method
- Breadcrumbs to take advantage of new logic introduced in latest G3 core for tags and search
- Added support for tag_albums module for pagination
- FP: Fix for Book frame pack - missing frame for movie thumbs
- Photo pages in Tag based collections do not support full item list. As result First/Last page paginator buttons and page list need to be disabled.
- Updated jQuery Cycle plug-in to latest version
- CSS: fixes for movie thumbnail in greydragon framepack

version 3.2.2
- CSS: Fixes a phenomenon where dialog content not aligning properly under some conditions
- CSS: Fixes a phenomenon where login dialog for error/login and reauthenticate pages were not in sync
- Movie thumb and Photo IDs are adjusted to comply with 3.0.3 notation
- Photo page image precache javascript moved into .ready to speed up the page load
- CP: Roundrobin styles improvements - top login positioning adjusted, removed padding for root page when sidebar is used
- ADMIN: Added prerequisite check for CURL support in PHP

version 3.2.1
- Logic to process command line parameters changed to prevent unhandled exception on some LAMP instances

version 3.2.0
- Theme is now HTML 5 
- CSS: Dropped support for IE 6
- IMPORTANT: .g-extended class moved from .g-item level into a <body> to allow layout manipulation based on thumb formats. Styles are affected.
- Added class marker to page's <body> element to indicate current sidebar mode - g-sidebar-xxxxxxxx
- CSS: Min width of the gallery is now enforced based on number of columns and sidebar visibility - extended thumbs 3/4 columns
- CSS:   Decrease size of the column for 3 column layout to 31% to prevent wrapping in some browser modes
- ADMIN: Logic added to inform site owner about new versions of the theme
- ADMIN: Admin page layout is now based on shared GreyDragon module which is now required 
- ADMIN: "Visible Title Length" option - access to G3 title length setting. 
- ADMIN: Use Permalinks for navigation - allows /root, /enter be used for root page access. 
  Requires the following rules added in .htaccess <mod_rewrite.c> section

  ########## Enable /root url ################
  #
  RewriteRule ^root ?root=yes [NC]
  RewriteRule ^enter ?root=no [NC]
  #
  ############################################

- Visible Title Length is now taken in consideration when displaying Photo Titles in thumbs, breadcrumbs, etc.
- Improved file name detection for Title Source
- CP: new color pack - Round Robin     \
- FP: new frame pack - Round Corners   / combined would give look of Clean Canvas theme
- FP: Book frame pack : removed right padding to prevent wrapping on smaller resolutions
- FP: Wall frame pack : min width decreased to all proper alignment of tiles
- jquery.cycle.js updated to v.2.9999.4
- Synchronize Calendar page with page.html.php
- CSS: forms.css - fixed two misspelled rules
- Added code protection from Camel case in CMD parameters
- Fixed issue with missing status icons for some colorpacks in Upload dialog
- Added code protection in Root page to detect situation when roller is empty and then do not initialize the cycle plugin
- Added translation support for default thumb caption (Photo #, Movie #, etc)
                                                                                                       
version 3.1.4
- ADMIN: Option to specify image quality level for Photo resizes and Thumbs
- ADMIN: Additional Description Display Mode for Photo page : Overlay Top (Static), Overlay Bottom (Static)

views/page.html.php
<?php defined("SYSPATH") or die("No direct script access.");
/**
 * G3 Grey Theme - a custom theme for Gallery 3
 * This theme is designed and built by David Yin, https://www.yinfor.com
 * Copyright (C) 2023 David Yin
 *
 * Based on the Grey Dragon Theme, which was designed and built by Serguei Dosyukov,
 * whose blog you will find at http://blog.dragonsoft.us/
 * Copyright (C) 2009-2014 Serguei Dosyukov
 *
 * This program is free software; you can redistribute it and/or modify it under the terms of the GNU General
 * Public License as published by the Free Software Foundation; either version 2 of the License, or (at your
 * option) any later version.
 *
 * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the
 * implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
 * for more details.
 *
 * You should have received a copy of the GNU General Public License along with this program; if not, write to
 * the Free Software Foundation, Inc., 51 Franklin Street - Fifth Floor, Boston, MA  02110-1301, USA.
 */
?>
<!DOCTYPE html >
<?php $theme->load_sessioninfo(); ?>
<html <?php echo $theme->html_attributes() ?> xml:lang="en" lang="en" <?php echo ($theme->is_rtl)? "dir=rtl" : null; ?> >
<?php
  $item = $theme->item();
  if (($theme->enable_pagecache) and (isset($item))):
    // Page will expire in 60 seconds
    header('Expires: ' . gmdate('D, d M Y H:i:s', time() + 60).'GMT');  
    header("Cache-Control: public");
    header("Cache-Control: post-check=3600, pre-check=43200", false);
    header("Content-Type: text/html; charset=UTF-8");
    header("Last-Modified: " . gmdate("D, d M Y H:i:s") . " GMT");
  endif;
?>
<!-- <?php echo $theme->themename ?> v.<?php echo $theme->themeversion ?> (<?php echo $theme->colorpack ?> : <?php echo $theme->framepack ?>) - Copyright (c) 2009-2014 Serguei Dosyukov - All Rights Reserved -->
<head>
<meta http-equiv="content-type" content="text/html; charset=UTF-8" />
<?php $theme->start_combining("script,css") ?>
<?php if ($page_title): ?>
<?php   $_title = $page_title ?> 
<?php else: ?>
<?php   if ($theme->item()): ?>
<?php     $_title = $theme->get_item_title($theme->item()); ?>
<?php   elseif ($theme->tag()): ?>
<?php     $_title = t("Photos tagged with %tag_title", array("tag_title" => $theme->bb2html($theme->tag()->name, 2))) ?>
<?php   else: /* Not an item, not a tag, no page_title specified.  Help! */ ?>
<?php     $_title = $theme->bb2html(item::root()->title, 2); ?>
<?php   endif ?>
<?php endif ?>
<title><?php echo $_title ?></title>
<meta name="title" content="<?php echo $_title ?>" /> 
<?php if ($theme->disable_seosupport): ?>
<meta name="robots" content="noindex, nofollow, noarchive" />
<meta name="googlebot" content="noindex, nofollow, noarchive, nosnippet, noodp, noimageindex, notranslate" />
<meta name="slurp" content="noindex, nofollow, noarchive, nosnippet, noodp, noydir" />
<meta name="teoma" content="noindex, nofollow, noarchive" />
<?php endif; ?>

<!-- Internet Explorer 9 Meta tags : Start -->
<meta name="application-name" content="<?php echo $_title; ?>" />
<meta name="msapplication-tooltip" content="<?php echo t("Start"); ?> <?php echo $_title; ?>" />
<meta name="msapplication-starturl" content="<?php echo item::root()->url() ?>" />                                                     
<?php if ($theme->allow_root_page): ?>
<meta name="msapplication-task" content="name=<?php echo t("Gallery") ?>: <?php echo t("Root Page") ?>; action-uri=<?php echo item::root()->url(); ?><?php echo $theme->permalinks["root"]; ?>; icon-uri=favicon.ico" />
<meta name="msapplication-task" content="name=<?php echo t("Gallery") ?>: <?php echo t("Root Album") ?>; action-uri=<?php echo item::root()->url(); ?><?php echo $theme->permalinks["enter"]; ?>; icon-uri=favicon.ico" />
<?php else: ?>
<meta name="msapplication-task" content="name=<?php echo t("Gallery") ?>: <?php echo t("Root Album") ?>; action-uri=<?php echo item::root()->url(); ?>; icon-uri=favicon.ico" />
<?php endif; ?>
<?php if (identity::active_user()->admin): ?>
<meta name="msapplication-task-separator" content="gallery3-g3grey" />
<meta name="msapplication-task" content="name=<?php echo t("Admin") ?>: <?php echo t("Dashboard") ?>; action-uri=<?php echo url::site("admin"); ?>; icon-uri=favicon.ico" />
<?php endif; ?>
<!-- Internet Explorer 9 Meta tags : End -->
<link rel="shortcut icon" href="<?php echo $theme->favicon ?>" type="image/x-icon" />
<?php if ($theme->appletouchicon): ?>
<link rel="apple-touch-icon" href="<?php echo $theme->appletouchicon; ?>"/>
<?php endif; ?>

<?php echo $theme->script("jquery.min.js"); ?>
<?php echo $theme->script("jquery.json.min.js"); ?>
<?php echo $theme->script("jquery.form.custom.js"); ?>
<?php echo $theme->script("jquery-ui.min.js"); ?>

<?php echo $theme->script("gallery.common.js") ?>
<?php /* MSG_CANCEL is required by gallery.dialog.js */ ?>
<script type="text/javascript">
  var MSG_CANCEL = <?php echo t('Cancel')->for_js() ?>;
</script>

<?php echo $theme->script("gallery.ajax.custom.js") ?>
<?php echo $theme->script("gallery.dialog.custom.js"); ?>

<?php /* These are page specific but they get combined */ ?>
<?php if ($theme->page_subtype == "photo"): ?>
<?php echo  $theme->script("jquery.scrollTo.js"); ?>
<?php elseif ($theme->page_subtype == "movie"): ?>
<?php echo  $theme->script("flowplayer.js") ?>
<?php endif ?>

<?php echo $theme->head() ?>

<?php // Theme specific CSS/JS goes last so that it can override module CSS/JS ?>
<?php echo $theme->theme_js_inject(); ?>
<?php echo $theme->theme_css_inject(); ?>
<?php echo $theme->get_combined("css");          // LOOKING FOR YOUR CSS? It's all been combined into the link ?>
<?php echo $theme->custom_css_inject(TRUE); ?>
<?php echo $theme->get_combined("script")        // LOOKING FOR YOUR JAVASCRIPT? It's all been combined into the link ?>

<?php if ($theme->thumb_inpage): ?>
<style type="text/css"> 
  #g-column-bottom #g-thumbnav-block, #g-column-top #g-thumbnav-block { display: none; } 
<?php if (((!$user->guest) or ($theme->show_guest_menu)) and ($theme->mainmenu_position == "bar")): ?>
  html { margin-top: 30px !important; }
<?php endif; ?>
</style>
<?php endif; ?>
</head>
<?php if ($theme->item()):
     $item = $theme->item();
   else:
     $item = item::root();
   endif; ?>
<body <?php echo $theme->body_attributes() ?><?php echo ($theme->show_root_page)? ' id="g-rootpage"' : null; ?> <?php echo $theme->get_bodyclass(); ?>>
<?php echo $theme->page_top() ?>                               
<?php echo $theme->site_status() ?>
<?php if (((!$user->guest) or ($theme->show_guest_menu)) and ($theme->mainmenu_position == "bar")): ?>
  <div id="g-site-menu" class="g-<?php echo $theme->mainmenu_position; ?>">
  <?php echo $theme->site_menu($theme->item() ? "#g-item-id-{$theme->item()->id}" : "") ?>
  </div>
<?php endif; ?>
<div id="g-header">
<?php echo $theme->header_top() ?>
<?php if ($theme->viewmode != "mini"): ?>
<?php   if ($header_text = module::get_var("gallery", "header_text")): ?>
<span id="g-header-text"><?php echo  $theme->bb2html($header_text, 1) ?></span>
<?php   else: ?>                                                                         
  <a id="g-logo" href="<?php echo item::root()->url() ?><?php echo ($theme->allow_root_page)? $theme->permalinks["root"] : null; ?>" title="<?php echo t("go back to the Gallery home")->for_html_attr() ?>">
    <img alt="<?php echo t("Gallery logo: Your photos on your web site")->for_html_attr() ?>" src="<?php echo $theme->logopath ?>" />
  </a>
<?php   endif; ?>
<?php endif; ?>
<?php if (((!$user->guest) or ($theme->show_guest_menu)) and ($theme->mainmenu_position != "bar")): ?>
  <div id="g-site-menu" class="g-<?php echo $theme->mainmenu_position; ?>">
  <?php echo $theme->site_menu($theme->item() ? "#g-item-id-{$theme->item()->id}" : "") ?>
  </div>
<?php endif ?>

<?php echo $theme->messages() ?>
<?php echo $theme->header_bottom() ?>

<?php if ($theme->loginmenu_position == "header"): ?>
<?php echo  $theme->user_menu() ?>
<?php endif; ?>
<?php if (empty($breadcrumbs)): ?>
<?php echo $theme->breadcrumb_menu(null); ?>
<?php else: ?>
<?php echo $theme->breadcrumb_menu($breadcrumbs); ?>
<?php endif; ?>
<?php echo $theme->custom_header(); ?>
</div>
<?php if (($theme->page_subtype != "login") and ($theme->page_subtype != "reauthenticate") and ($theme->sidebarvisible == "top")): ?>
<div id="g-column-top">
  <?php echo new View("sidebar.html") ?>
</div>
<?php endif; ?>
<div id="g-main">
  <div id="g-main-in">
<?php  if (!$theme->show_root_page): ?>
    <?php echo $theme->sidebar_menu($item->url()) ?>
    <div id="g-view-menu" class="g-buttonset<?php echo ($theme->sidebarallowed != "any")? " g-buttonset-shift" : null; ?>">
<?php    if ($page_subtype == "album"): ?>
      <?php echo $theme->album_menu() ?>
<?php    elseif ($page_subtype == "photo") : ?>
      <?php echo $theme->photo_menu() ?>
<?php    elseif ($page_subtype == "movie") : ?>
      <?php echo $theme->movie_menu() ?>
<?php    elseif ($page_subtype == "tag") : ?>
      <?php echo $theme->tag_menu() ?>
<?php    endif ?>
    </div>
<?php  endif; ?>
<?php switch ($theme->sidebarvisible):
     case "left":
       echo '<div id="g-column-left">';
       $closediv = TRUE;
       break;
     case "none":
     case "top":
     case "bottom":
       if (($theme->thumb_inpage) and ($page_subtype == "photo")):
         echo '<div id="g-column-right">';
         $closediv = TRUE;
       else:
         $closediv = FALSE;
       endif;
       break;
     default:
       echo '<div id="g-column-right">';
       $closediv = TRUE;
       break;
   endswitch; ?>
<?php if (($theme->page_subtype != "login") && ($theme->page_subtype != "reauthenticate")): ?>
<?php   if (($theme->sidebarvisible == "none") || ($theme->sidebarvisible == "bottom") || ($theme->sidebarvisible == "top")): ?>
<?php     if (($theme->thumb_inpage) and ($page_subtype == "photo")): ?>
<?php echo      '<div class="g-toolbar"><h1>&nbsp;</h1></div>'; ?>
<?php echo      $theme->get_block_html("thumbnav"); ?>
<?php     endif; ?>
<?php   else: ?>
<?php echo    new View("sidebar.html") ?>
<?php   endif; ?>
<?php endif ?>
<?php echo ($closediv)? "</div>" : null; ?>

<?php switch ($theme->sidebarvisible):
     case "left":
       echo '<div id="g-column-centerright">';
       break;
     case "none":
     case "top":
     case "bottom":
       if (($theme->thumb_inpage) and ($page_subtype == "photo")):
         echo '<div id="g-column-centerleft">';
       else:
         echo '<div id="g-column-centerfull">';
       endif;
       break;
     default:
       echo '<div id="g-column-centerleft">';
       break;
   endswitch;

   if ($theme->show_root_page):
     echo new View("rootpage.html");
   else:
     echo $content;
   endif; ?>
    </div> 
  </div>
</div>
<?php if (($theme->page_subtype != "login") and ($theme->page_subtype != "reauthenticate") and ($theme->sidebarvisible == "bottom")): ?>
<div id="g-column-bottom">
  <?php echo new View("sidebar.html") ?>
</div>
<?php endif; ?>
<div id="g-footer">
<?php if ($theme->viewmode != "mini"): ?>
<?php echo  $theme->footer() ?>
<?php   if ($footer_text = module::get_var("gallery", "footer_text")): ?>
<span id="g-footer-text"><?php echo  $theme->bb2html($footer_text, 1) ?></span>
<?php   endif ?>
  <?php echo $theme->credits() ?>
  <ul id="g-footer-rightside"><li><?php echo $theme->copyright ?></li></ul>
<?php   if ($theme->loginmenu_position == "default"): ?>
  <?php echo $theme->user_menu() ?>
<?php   endif; ?>
<?php endif; ?>
<?php echo $theme->custom_footer(); ?>
</div>
<?php echo $theme->page_bottom() ?>
</body>
</html>
